Microsoft.Toolkit.Uwp.UI.Controls/TextToolbar/Formats/RichText/RichTextButtonActions.cs

```diff
@@ -0,0 +1,147 @@
+        {
+            var format = Formatter.SelectionFormat;
+            if (!button.IsToggled)
+            {
+                format.Italic = FormatEffect.On;
+                Formatter.SelectionFormat = format;
+            }
+            else
+            {
+                format.Italic = FormatEffect.Off;
+                Formatter.SelectionFormat = format;
+            }
+
+            button.IsToggled = button.IsToggled != true;
+        }
+
+        public override void FormatStrikethrough(ToolbarButton button)
+        {
+            var format = Formatter.SelectionFormat;
+            if (!button.IsToggled)
+            {
+                format.Strikethrough = FormatEffect.On;
+                Formatter.SelectionFormat = format;
+            }
+            else
+            {
+                format.Strikethrough = FormatEffect.Off;
+                Formatter.SelectionFormat = format;
+            }
+
+            button.IsToggled = button.IsToggled != true;
+        }
+
+        public override void FormatLink(ToolbarButton button, string label, string formattedText, string link)
+        {
+            var selected = Formatter.Selected;
+            if (!string.IsNullOrWhiteSpace(label))
+            {
+                selected.SetText(TextSetOptions.FormatRtf, formattedText);
+            }
+            else
+            {
+                selected.Text = link;
+            }
+
+            selected.Link = $"\"{link}\"";
+
+            var doc = Formatter.Model.Editor.Document;
+            doc.ApplyDisplayUpdates(); // doesn't seem to work
+        }
+
+        public override void FormatList(ToolbarButton button)
+        {
+            var selected = Formatter.Selected;
+            if (!button.IsToggled)
+            {
+                Formatter.OrderedListButton.IsToggled = false;
+                selected.ParagraphFormat.ListType = MarkerType.Bullet;
+                selected.ParagraphFormat.ListStyle = MarkerStyle.Plain;
+            }
+            else
+            {
+                selected.ParagraphFormat.ListType = MarkerType.None;
+            }
+
+            button.IsToggled = button.IsToggled != true;
+        }
+
+        public override void FormatOrderedList(ToolbarButton button)
+        {
+            var selected = Formatter.Selected;
+            if (!button.IsToggled)
+            {
+                Formatter.ListButton.IsToggled = false;
+                selected.ParagraphFormat.ListType = MarkerType.UnicodeSequence;
+                selected.ParagraphFormat.ListStyle = MarkerStyle.Period;
+            }
+            else
+            {
+                selected.ParagraphFormat.ListType = MarkerType.None;
+            }
+
+            button.IsToggled = button.IsToggled != true;
+        }
+
+        public void FormatUnderline(ToolbarButton button)
+        {
+            var format = Formatter.SelectionFormat;
+            if (!button.IsToggled)
```

Comment:
Some of the properties on these a very slightly different though, such as Underline and list types.